### PR TITLE
PD-779: Configure .design Environment

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

This PR creates website dir for new mongodb.design. Next and emotion are installed, so that we can be up and running after we configure this skeleton with Realm. Our eslint config is now formatting files contained within this directory, but I created a new .tsconfig local to the website directory, because some of our rules in package.tsconfig.json are in conflict with rules that next.js requires (noEmit: "true").

To note: When testing locally, I discovered a bug with our `run init` script. Before `cd`-ing into the `website` dir and running `yarn dev` first run `yarn build` from our root. Ticket here: [Investigate build script](https://jira.mongodb.org/browse/PD-830)

🎟 _Jira ticket:_ [PD-779](https://jira.mongodb.org/browse/PD-779)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 node_modules
 dist
 lib
-storybook/public
\ No newline at end of file
+storybook/public
+website/.next
```
Comment (brookescarlett) [inline .eslintignore:5]: This doesn't seem to be working, still seeing these files appear when I run `yarn format` 🤔

File: website/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "compilerOptions": {
+    "rootDir": "pages",
+    "jsx": "preserve",
+    "lib": ["dom", "dom.iterable", "esnext"],
+    "skipLibCheck": true,
+    "forceConsistentCasingInFileNames": true,
+    "esModuleInterop": true,
+    "resolveJsonModule": true,
+    "noEmit": true,
```
Comment (brookescarlett) [inline website/tsconfig.json:10]: This rule is required by next.js and conflicts with our existing package.tsconfig, which is why I'm not extending our rules here.

File: .eslintignore (hunk 1 above)
Comment (hswolff) [reply .eslintignore:5]: May need to add it to `.prettierignore` as well.

File: .eslintrc.js [hunk 3]
```diff
@@ -26,6 +26,7 @@ module.exports = {
     },
   },
   rules: {
+    'react/react-in-jsx-scope': 'off',
```
Comment (hswolff) [inline .eslintrc.js:29]: We don't want to apply this to every file right, just the files in website? We can use the overrides config to be more selective in what this rule applies to.

Comment (brookescarlett) [reply .eslintrc.js]: 👍

File: .eslintignore (hunk 1 above)
Comment (brookescarlett) [reply .eslintignore:5]: 🤦

Comment (brookescarlett) [reply .eslintignore:5]: updated

File: website/tsconfig.json (hunk 2 above)
Comment (hswolff) [reply website/tsconfig.json:10]: 😬

**Pull request merged**